flytekit/models/core/errors.py

```diff
@@ -49,19 +60,40 @@ def origin(self) -> int:
         """
         return self._origin
 
+    @property
+    def timestamp(self) -> Timestamp:
+        """
+        The timestamp of the error, as number of seconds and nanos since Epoch
+        """
+        return self._timestamp
+
+    @property
+    def worker(self) -> int:
+        """
+        The worker name where the error originated
+        """
+        return self._worker
+
     def to_flyte_idl(self):
         """
         :rtype: flyteidl.core.errors_pb2.ContainerError
         """
-        return _errors_pb2.ContainerError(code=self.code, message=self.message, kind=self.kind, origin=self.origin)
+        return _errors_pb2.ContainerError(
+            code=self.code,
+            message=self.message,
+            kind=self.kind,
+            origin=self.origin,
+            timestamp=self._timestamp,
+            worker=self.worker,
+        )
 
     @classmethod
     def from_flyte_idl(cls, proto):
         """
         :param flyteidl.core.errors_pb2.ContainerError proto:
         :rtype: ContainerError
         """
-        return cls(proto.code, proto.message, proto.kind, proto.origin)
+        return cls(proto.code, proto.message, proto.kind, proto.origin, proto.timestamp, proto.worker)
```

Comment:
there's some pretty basic tests for these model files that we've been trying to get rid of unsuccessfully forever here: https://github.com/flyteorg/flytekit/blob/master/tests/flytekit/unit/models/core/test_errors.py

while we have them, mind adding the new fields to that test there?

Reply:
Sure, adding.